### PR TITLE
YQL-17087: Add channel spilling to dq pipe communication

### PR DESCRIPTION
**Pull request by Darych**

### CODE REVIEW

File: ydb/library/yql/dq/actors/spilling/channel_storage_actor.h [hunk 1]
```diff
@@ -5,14 +5,32 @@
...
 
-class IDqChannelStorageActor : public IDqChannelStorage
+class IDqChannelStorageActor
 {
 public:
-    virtual void Terminate() = 0;
+    using TPtr = TIntrusivePtr<IDqChannelStorageActor>;
+    using TWakeUpCallback = std::function<void()>;
+
+    virtual ~IDqChannelStorageActor() = default;
```
Comment (lll-phill-lll) [inline ydb/library/yql/dq/actors/spilling/channel_storage_actor.h:14]: 👍

File: ydb/library/yql/providers/dq/task_runner_actor/task_runner_actor.cpp [hunk 2]
```diff
@@ -89,10 +120,69 @@ class TOutputChannelReader {
...
+        bool changed = false;
+        bool isChanFinished = false;
+        i64 remain = ToPopSize;
+        ui32 dataSize = 0;
+        bool hasData = true;
+        TOutputChannelReadResult result;
+
+        if (remain == 0) {
+            // special case to WorkerActor
+            remain = 5<<20;
```
Comment (lll-phill-lll) [inline ydb/library/yql/providers/dq/task_runner_actor/task_runner_actor.cpp:134]: Is it possible to use `5_MB` here? Like in this line https://github.com/ydb-platform/ydb/pull/612/files?diff=unified&w=0#diff-0755ee9481d9c53853e4dc79733dd98ddf730219572b1cf5501dac5f3e285e41R35

File: ydb/library/yql/providers/dq/task_runner_actor/task_runner_actor.cpp [hunk 3]
```diff
@@ -89,10 +120,69 @@ class TOutputChannelReader {
...
+            maxChunks = 1;
+        }
+
+        auto spillingStorage = SpillingStorageInfo->SpillingStorage;
+        // Read all available data from the pipe and spill it
+        for (;spillingStorage && !isChanFinished && hasData;) {
```
Comment (lll-phill-lll) [inline ydb/library/yql/providers/dq/task_runner_actor/task_runner_actor.cpp:140]: while?

**Review by lll-phill-lll — APPROVED** [review]

**Review by resetius — APPROVED** [review]

**Review by rvu1024 — APPROVED** [review]

**Review by resetius — APPROVED** [review]
ok

**Pull request merged**